### ISSUE TITLE
String interpolation produces incorrect output for concatenated strings

### ISSUE THREAD

**Issue by hartez** [dotnet/roslyn#55569]
### Description

Concatenating interpolated strings is producing incorrect output using the latest preview. For example:

```
var a = 0;
var b = 1;
var c = 3;

var concatenated = $"{a}: " +
					$"{b} " +
					$"{c}";

var single = $"{a}: {b} {c}";

Console.WriteLine(concatenated);
Console.WriteLine(single);
```

Expected output:
```
0: 1 3
0: 1 3
```

Actual output:
```
: 0 13
0: 1 3
```

### Configuration

* 6.0.100-preview.7.21379.14
* Windows 10 x64
* I'm not certain if this is limited to this OS and architecture

### Regression?

Yes; this worked as expected in .NET 5 and earlier .NET 6 previews.

**Comment by viniciusverasdossantos** [comment]
I installed SDK 6.0.0-rc.1.21418.9 and the problem continues

**Comment by jcouv** [comment]
@viniciusverasdossantos Could you confirm what compiler version you used in latest repro? (what compiler version is included in SDK 6.0.0-rc.1.21418.9?)
To do this, add `#error version` to your source and share the resulting error message. I'll be able to check whether that compiler version included the fix we've already made.

**Comment by jcouv** [comment]
Thanks @ufcpp We indeed have tests for this now. I'll go ahead and close
